Subject: [PATCH] Fix NPE when call stack.setLauncheTime.

The stack of task maybe null while call ActivityStack#setLaunchTime, should check
the stack of activity record before call setLaunchTime, avoid the NPE leads
system_server process to crash.

diff --git a/services/core/java/com/android/server/am/ActivityStackSupervisor.java b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
index 2262697..63a0ee4 100644
--- a/services/core/java/com/android/server/am/ActivityStackSupervisor.java
+++ b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
@@ -1397,7 +1397,12 @@
         ProcessRecord app = mService.getProcessRecordLocked(r.processName,
                 r.info.applicationInfo.uid, true);
 
-        r.task.stack.setLaunchTime(r);
+        if (r.task != null && r.task.stack != null) {
+            r.task.stack.setLaunchTime(r);
+        } else {
+            Slog.w(TAG, "Stack or task of activity:" + r
+                    + " is null, will not setLaunchTime for it.");
+        }
 
         if (app != null && app.thread != null) {
             try {
